### ISSUE TITLE
Error when using "deploy to Azure" from quickstarts page. The template doesn't validate.

### ISSUE THREAD

**Issue by b3nt0** [Azure/azure-quickstart-templates#5960]
--------------------MESSAGE FROM ADMIN, DELETE BEFORE SUBMITTING----------------------

Sorry to hear you had a bad experience with one of the templates :worried: But, in case you're just asking a question, we're happy to help. You can also check if the question might already have been asked here https://github.com/Azure/azure-quickstart-templates/issues?utf8=%E2%9C%93&q=is%3Aissue

We've created an outline of recommended sections to fill out that will help make this Pull Request awesome!

--------------------MESSAGE FROM ADMIN, DELETE BEFORE SUBMITTING----------------------

[Template Name goes here](Template link goes here)

### Issue Details
{
  "telemetryId": "d41b85fb-e538-4211-9b73-3093c23722b0",
  "bladeInstanceId": "Blade_0fae8553579049e8b9811d2f7d96ee7f_0_0",
  "galleryItemId": "Microsoft.Template",
  "createBlade": "DeployToAzure",
  "code": "InvalidTemplate",
  "message": "Deployment template validation failed: 'The template resource 'spfarm-ad/InstallDomainController' for type 'extensions' at line '97' and column '9' has incorrect segment lengths. A nested resource type must have identical number of segments as its resource name. A root resource type must have segment length one greater than its resource name. Please see https://aka.ms/arm-template/#resources for usage details.'."
}

### Repro steps (*if necessary, delete otherwise*)
1. Go to https://azure.microsoft.com/en-us/resources/templates/sharepoint-three-vm/
2. Click "deploy to Azure"
3. Add parameter values where necessary.
4. click "Purchase"
5. Get validation error.

**Comment by Yvand** [comment]
Did you consider https://azure.microsoft.com/en-us/resources/templates/sharepoint-adfs/ instead?

**Comment by rohit404404** [comment]
> Did you consider https://azure.microsoft.com/en-us/resources/templates/sharepoint-adfs/ instead?

Yes. However this creates heavy resources and costs goes beyond 540 . Three VM is relatively less resource intensive. I am facing same issue above and not sure whats wrong. Dont see any segment length issue.

**Comment by Yvand** [comment]
@rohit404404 template https://azure.microsoft.com/en-us/resources/templates/sharepoint-adfs/ creates only 3 VMs, I think you are confusing with template https://azure.microsoft.com/en-us/resources/templates/sharepoint-server-farm-ha/, which indeed creates a lot more VMs

**Comment by rohit404404** [comment]
@Yvand  Great. Your template worked perfectly. I faced issue initially because i chose Asia location and said Storage type was not available. I moved to US and it worked perfectly.

Now i just need to find a way to disable all the services properly to avoid unnecessary cost. I do have autoshutdown turned on, however concerned about other resources. If there was a script to start and stop all resoruces whenver needed :P. 
 Will see what bill i get for two days. I just have 100 euro balance that i get every month with MSDN

Thank you again!!

**Comment by Yvand** [comment]
Thank you for your feedback!
What Azure service(s) are you looking to disable? None is really optional

**Comment by rohit404404** [comment]
@Yvand  I meant, I need these machine only when I need to do some POCs, else they are idle mostly. However there is still a disk cost.
SO it would have been great to have SQL and DC on single server and SharePoint on another server, that way i can avoid cost of one disk. Any pointers on that please?

**Comment by Yvand** [comment]
I see, but I know no such template
